src/components/datetime/DateTimeController.cpp

```diff
@@ -75,6 +75,22 @@ void DateTime::UpdateTime(uint32_t systickCounter) {
   minute = time.minutes().count();
   second = time.seconds().count();
 
+  if (minute == 0 and not isHourAlreadyNotified) {
+    isHourAlreadyNotified = true;
+    if (systemTask != nullptr)
+      systemTask->PushMessage(System::Messages::OnNewHour);
+  } else if (minute != 0) {
+    isHourAlreadyNotified = false;
+  }
+
+  if ((minute == 0 or minute == 30) and not isHalfHourAlreadyNotified) {
```

Comment:
Usage of `and` and `not` is not consistent with the rest of the code.

Reply:
Is this preferred please?
`if ((minute == 0 or minute == 30) && !isHalfHourAlreadyNotified)`

Reply:
Yep, that's good. Don't forget the closing brace though :D

Reply:
Should I also replace the `or` with `||` too?

Reply:
That would be more consistent, yes.